spec/datadog/core/telemetry/metric_spec.rb

```diff
@@ -0,0 +1,184 @@
+require 'spec_helper'
+
+require 'datadog/core/telemetry/metric'
+
+RSpec.describe Datadog::Core::Telemetry::Metric::Count do
+  subject(:metric) { described_class.new('tests', { foo: :bar }) }
+  let(:generate_metric_type) { 'generate-metrics' }
+
+  describe '.request_type' do
+    subject(:request_type) { metric.class.request_type }
+
+    it { is_expected.to eq(generate_metric_type) }
```

Comment:
Minor: Since this is only used once, consider inlining it? (Same applies to other examples in this file)

```suggestion
  describe '.request_type' do
    subject(:request_type) { metric.class.request_type }

    it { is_expected.to eq('generate-metrics') }
```